src/smallstack.rs

```diff
@@ -92,33 +92,31 @@ impl<'a> VM<'a> {
         &["carry", "invert"] => carry = !carry,
 
         &["call", "primitive", "say"] => if let Some(Expr::Str(arg)) = stack.pop() {
-          println!("hey {}", arg);
+          println!("{}", arg);
         } else {
           panic!("VM error: incorrect arguments to `say`");
         },
 
+        &["call", "primitive", "concat"] => if let
+            (Some(Expr::Str(arg1)), Some(Expr::Str(arg2)))
+            =
+            (stack.pop(), stack.pop()) {
+          stack.push(Expr::Str(arg1 + &arg2));
```

Comment:
somewhat feeling uneasy about this.

i guess we should have `&["call", "primitive", x] => ` and then try to find the function in some map (the stdlib).